src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs

```diff
@@ -819,7 +819,7 @@ private void UpdateInlines(string text)
 				return;
 			}
 
-			if (ReadLocalValue(TextProperty) is UnsetValue)
+			if (ReadLocalValue(TextProperty) == DependencyProperty.UnsetValue)
```

Comment:
Should we be using ReferenceEquals instead and avoid calling the Equals override on the incoming object?

Reply:
I think this will be compiled into a ReferenceEquals call because the other side of the `==` has compile-time type of `object`. So, our `==` operator isn't a candidate:

https://github.com/unoplatform/uno/blob/0fb80f819fb57b37476edff7c3a174e93640f495/src/Uno.UI/UI/Xaml/UnsetValue.cs#L26

Reply:
![image](https://github.com/unoplatform/uno/assets/31348972/8a0c65e5-3805-42ea-aedd-3426ded406bc)